### ISSUE TITLE
UnnecessaryParentheses: Allow to use parenthesis when in math expression with mixed operators

### ISSUE THREAD

**Issue by marianosimone** [arturbosch/detekt#969]
Even though they are not necessary (for a computer), parentheses make life easier (for humans).

```
val hardToRead = 1 + 3 / 2
val easierToRead = 1 + (3 / 2)
```

## Expected Behavior
The `easierToRead` expression shouldn't trigger `UnnecessaryParentheses`

## Current Behavior
The `hardToRead` option is proposed

**Comment by schalkms** [comment]
I don't think that this is a good idea for the following reason.
One can argue the same for an if condition for instance. If one prefers readability over smaller code size this rule should be deactivated.

**Comment by marianosimone** [comment]
I think one of the problems with this rule is that it detects different things:

- Parentheses that are unnecessary for arithmetic/logic reasons (math operations and boolean algebra)
- Parentheses that are unnecessary for syntactic reasons (from the examples: `function({ input -> println(input) })`)

I can see people wanting the second group being reported, but not the first. Currently, there's no way to only disable one side, and that's what I think would be valuable (either by making this rule configurable, or by splitting it up)

I can see an argument being made for things like: `if ((a == b)) { }`, as it's a boolean expression that can be simplified, but it would be OK to trigger just because there's no boolean operator involved

**Comment by schalkms** [comment]
I think splitting up the rule makes more sense.
I'm okay with having two `UnnecessaryParentheses*` rules, as you mentioned.

**Comment by Mauin** [comment]
The `UnnecessaryParentheses` rule was inspired by the same check that IntelliJ offers. Looking at IntelliJ's behavior it will also point out the unnecessary parentheses in the `easierToRead` case. 

I'm not saying I disagree with adding more options to this rule, but currently it should be reporting (hopefully) the same issues that also IntelliJ is pointing out.

**Comment by schalkms** [comment]
Thanks for pointing that out, Marvin. Detekt's rules comply with IntelliJ's. We should do the same here. I'll close this.

**Comment by alexbchr** [comment]
It would be great if we had some customizations too for this analysis. Another example, similar to the `easierToRead` Math expression, is with logic conditions.
```kotlin
val easierToRead = (a && b) || (c && d)
val harderToRead = a && b || c && d
```

The analysis flag the easierToRead option, suggesting to use the harderToRead option, which is not really convenient. It would be great to have more customization for this analysis, because I really want to use it for certain cases, such as lambda parentheses, but not for certain things such as math and logic conditions.

**Comment by arturbosch** [comment]
hey @alexbchr feel free to open a new issue with your usecase.
